### DISCUSSION TITLE
flux create helmrelease not pushing to git provider, is this expected?

### DISCUSSION THREAD

**Discussion by crashingmotorbikes** [fluxcd/flux2#3253] [Q&A] [answered]
According to the Flux documentation using flux create helmrelease should be committing to version control the CRDs as instructions. (https://fluxcd.io/flux/use-cases/helm/).

I'm not seeing any changes within my GitLab repo where flux has bootstrapped itself though. I do see that --export can be used to get the generated files to be able to push to version control too.

I'm wondering what I may be missing here. Should 'flux create helmrelease' be committing to the remote git repository or do I have to do that manually via --export, git add/commit/push? If this is the case I think it would be helpful for the documentation to be updated with an example of what an 'instruction' means and that the git provider will not receive anything about the flux resources.

Note:
My helmrelease has been deployed correctly and the endpoint for my service is accessible.
I'm running 'flux create helmrelease' via Gitlab pipeline but have also performed it manually with the same results.

**Comment by georgek** [comment]
Same problem, not sure if I've missed something. Same with `flux create helmrelease`. It applies the changes imperatively to the cluster, but I'm not seeing a manifest created anywhere unless I do `--export` and manually commit it to the repo.

**Comment by stefanprodan** [comment]
No Flux command pushes to Git expect for `flux bootstrap`. All the `flux create` commands have an `--export` flag and it's up to you where you want this stored in Git. I recommend adopting a repository structure where HelmReleases can be patched per environment (dev/stg/prod). There are many ways of structuring repos for GitOps, if you're new to Flux, I suggest going with the example repo: https://github.com/fluxcd/flux2-kustomize-helm-example

**Reply by georgek** [reply, marked as answer]
Thanks, for the reply. In that case, I think the docs are a bit unclear or wrong in this respect. Specifically the page linked above (https://fluxcd.io/flux/use-cases/helm/) says: 

> The main difference is that the Flux client will not imperatively create resources in the cluster.

But in fact, this command *does* imperatively create resources on the cluster. It would seem it would only do what is described with the `--export` flag. The note about `--export` is even more misleading:

> note: you may use the --export flag to manage any file edits with finer grained control before pushing to version control

Personally I found this `flux create` quite unintuitive as I thought the whole point of flux was not to do stuff imperatively and never have infrastructure that isn't under version control. In other words, I would have expected `--export` to be the default behaviour.

**Reply by stefanprodan** [reply]
Good point, the create commands are toys to play around, I don’t think any Flux user relies on them. Will update the docs
